website/static/js/addProjectPlugin.js

```diff
@@ -176,12 +176,13 @@ var AddProject = {
                             m('label[for="projectName].f-w-lg.text-bigger', 'Title'),
                             m('input[type="text"].form-control.project-name', {
                                 onkeyup: function(ev){
-                                    if (ev.which === 13) {
-                                         ctrl.add();
-                                    }
                                     var val = ev.target.value;
+                                    var validTitle = val.trim().length > 0;
+                                    if (ev.which === 13 && validTitle) {
```

Comment:
I think this works, but I think it could probably be a tiny bit better. I would lean towards setting `ctrl.isValid()` early, then doing the `ctrl.add()` if it's return after that. Instead (or possibly in addition to) the validity check here for whether to `add()`, I would definitely put at the top of `add()` the check for title validity. If it's there, then as long as validity is being verified properly here, then no matter what rout the add gets called, it will have to have a valid title in order to add, and if it's close enough to to the top of the function, then the overhead for checking there vs. here is negligible. This future-proofs in case someone wants to put a new UI control that will call `add()`.